go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -438,11 +439,15 @@ func (ncc *networkClusterController) Cleanup() error {
 }
 
 func (ncc *networkClusterController) Reconcile(netInfo util.NetInfo) error {
-	// update network informaiton, point of no return
+	reconcilePendingPods := !ncc.ReconcilableNetInfo.EqualNADs(netInfo.GetNADs()...)
+	// update network information, point of no return
 	err := util.ReconcileNetInfo(ncc.ReconcilableNetInfo, netInfo)
 	if err != nil {
 		klog.Errorf("Failed to reconcile network %s: %v", ncc.GetNetworkName(), err)
 	}
+	if reconcilePendingPods {
+		return objretry.RequeuePendingPodsForNetwork(ncc.kube, ncc.GetNetInfo(), ncc.retryPods)
```

Comment:
Would just log this error.

Returning this error would cause a noop retry on which `reconcilePendingPods` would no longer be true and nothing would be done.

The real issue lies on the retry framework, re-queue should just not fail.

Reply:
Same for other instances.